library/IReplicatedTable.cs

```diff
@@ -20,32 +20,32 @@
 
 namespace Microsoft.Azure.Toolkit.Replication
 {
+    using System;
     using System.Collections.Generic;
-    using Microsoft.WindowsAzure.Storage;
-    using Microsoft.WindowsAzure.Storage.Table;
+    using System.Linq.Expressions;
+    using global::Azure;
+    using global::Azure.Data.Tables;
 
     public interface IReplicatedTable
     {
         string TableName { get; }
-        bool CreateIfNotExists(TableRequestOptions requestOptions = null, OperationContext operationContext = null);
+        bool CreateIfNotExists();
         bool Exists();
         bool DeleteIfExists();
-        TableResult Execute(TableOperation operation, TableRequestOptions requestOptions = null, OperationContext operationContext = null);
-        IList<TableResult> CheckRetrieveInBatch(TableBatchOperation batch, TableRequestOptions requestOptions = null, OperationContext operationContext = null);
-        IList<TableResult> ExecuteBatch(TableBatchOperation batch, TableRequestOptions requestOptions = null, OperationContext operationContext = null);
-        TableResult Retrieve(TableOperation operation, TableRequestOptions requestOptions = null, OperationContext operationContext = null);
-        TableResult Delete(TableOperation operation, TableRequestOptions requestOptions = null, OperationContext operationContext = null);
-        TableResult Merge(TableOperation operation, TableResult retrievedResult, TableRequestOptions requestOptions = null, OperationContext operationContext = null);
-        TableResult InsertOrMerge(TableOperation operation, TableRequestOptions requestOptions = null, OperationContext operationContext = null);
-        TableResult Replace(TableOperation operation, TableResult retrievedResult, TableRequestOptions requestOptions = null, OperationContext operationContext = null);
-        TableResult Insert(TableOperation operation, TableResult retrievedResult, TableRequestOptions requestOptions = null, OperationContext operationContext = null);
-        TableResult InsertOrReplace(TableOperation operation, TableRequestOptions requestOptions = null, OperationContext operationContext = null);
-        TableResult FlushAndRetrieve(IReplicatedTableEntity row, TableRequestOptions requestOptions = null, OperationContext operationContext = null, bool virtualizeEtag = true);
-        IEnumerable<TElement> ExecuteQuery<TElement>(TableQuery<TElement> query, TableRequestOptions requestOptions = null, OperationContext operationContext = null) where TElement : ITableEntity, new();
-        TableQuery<TElement> CreateQuery<TElement>()
-            where TElement : ITableEntity, new();
-        ReplicatedTableQuery<TElement> CreateReplicatedQuery<TElement>()
-            where TElement : ITableEntity, new();
+        IList<TableResult> ExecuteBatch(IEnumerable<TableTransactionAction> batch);
```

Comment:
Need more context on why the changes in the interface here, especially removal of API like Execute which is used by callers

Reply:
Execute relies on passing TableOperation object which does not exist in new SDK. Instead of clients calling Execute, they can directly call the requested operation: Retrieve, Insert, Replace, InsertOrReplace, Delete

Changes in client will be like the following 
![image](https://github.com/Azure/rtable/assets/122410443/0f72abf4-0423-4b92-b642-5ae540e626fb)